pkg/api/types.go

```diff
@@ -421,6 +425,19 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
```

Comment:
I can buy that argument for persistent volumes, where it is an admin that is crossing the namespace boundary.  It's a bit less shiny when it's a user's pod that is referencing Endpoints or Services in another namespace.

As it stands, the endpoints must be in the same namespace as the pod.  I don't think this is sufficient to handle what you are describing.  This should probably become an ObjectRef, or else we should make it target a multi-record DNS name and treat that as an endpoints set (or something).

Then we have to decide if it is kosher to write an Endpoints object that does not have an associated Service object.